models/ac_generator.cpp

```diff
@@ -153,16 +206,33 @@ nest::ac_generator::calibrate()
 void
 nest::ac_generator::update( Time const& origin, const long from, const long to )
 {
+  assert(
+    to >= 0 && ( delay ) from < kernel().connection_manager.get_min_delay() );
+  assert( from < to );
+
   long start = origin.get_steps();
 
-  CurrentEvent ce;
   for ( long lag = from; lag < to; ++lag )
+  {
+    B_.logger_.record_data( origin.get_steps() + lag );
+    S_.I_ = 0.0;
+
     if ( device_.is_active( Time::step( start + lag ) ) )
     {
       const double y_0 = S_.y_0_;
       S_.y_0_ = V_.A_00_ * y_0 + V_.A_01_ * S_.y_1_;
       S_.y_1_ = V_.A_10_ * y_0 + V_.A_11_ * S_.y_1_;
+      S_.I_ = S_.y_1_ + P_.offset_;
+
+      CurrentEvent ce;
```

Comment:
I had moved it inside, thinking that the object be constructed only if it was going to be used (device active). This, I think, is also how it was coded in [sinusoidal_poisson_generator.cpp](https://github.com/nest/nest-simulator/blob/master/models/sinusoidal_poisson_generator.cpp#L275)